platforms/ide/tooling-api-builders/src/test/groovy/org/gradle/tooling/internal/provider/runner/DefaultBuildControllerTest.groovy

```diff
@@ -224,39 +223,14 @@ class DefaultBuildControllerTest extends Specification {
         _ * workerThreadRegistry.workerThread >> true
         1 * modelController.runQueryModelActions(_) >> { def params ->
             def actions = params[0]
-            actions.forEach { it.run(null) }
+            actions.collect { it.get() }
         }
         1 * action1.get() >> "one"
         1 * action2.get() >> "two"
         1 * action3.get() >> "three"
         0 * _
     }
 
-    def "collects all failures from actions"() {
```

Comment:
ℹ️  Since there are two levels of indirection now from the `DefaultBuildController`, the unit test has been moved to the `BuildModelNestedActionRunner` test.